### PR TITLE
Port pg_graphql to rust

### LINKED ISSUE #133: `numeric` declares as `String` in graphql Schema but serializes as a `Float`
Add support for `numeric`

### LINKED ISSUE #170: Queries with "after" always returns "hasPreviousPage: true"
**Describe the bug**
Attempting to create a paginated list of results I tried to query for the `pageInfo`.
To conditionally show a `Previous`/`Next` button in the UI I wanted to use the `hasPreviousPage`/`hasNextPage` values.
However in attempting to pass the offset filter to the query I noticed that the `hasPreviousPage ` is always true, no matter the value. This is true for:
- `after: null`
- `after: $firsNodeCursor`

Let me know if more context is needed,
Thanks!

**To Reproduce**
Query a collection's `pageInfo` and attempt to create an offset using `after`

```gql
query {
  orgsCollection {
    edges {
      node {
        entries(
          first: 2,
          after: null
        ) {
          totalCount
          pageInfo {
            hasNextPage
            hasPreviousPage
            startCursor
            endCursor
          }
        }
      }
    }
  }
}
```

**Expected behavior**
`hasPreviousPage` should be `false` if the value for the `after` cursor is `null` or the cursor for the first node in the results.

**Screenshots**
#### ✅ Correct Behaviour without `after` argument
Screenshot 2022-04-26 at 15 22 16

#### ❌ Incorrect with `after: null`
Screenshot 2022-04-26 at 15 23 38

#### ❌ Incorrect with `after` set to the first node's cursor (see first screenshot)
Screenshot 2022-04-26 at 15 22 59

**Versions:**
 - PostgreSQL: 14.2
 - pg_graphql commit ref: 0.2.0 (Latest included in Supabase image)

### LINKED ISSUE #183: First resolution of large query is slow
**Describe the bug**
We have some queries that request a lot of nested data through various relationships and pivot tables. When running these large queries in our app (aka going via other pages which run different queries) they can often take a few seconds to return data, but if we hit one repeatedly (e.g. in a load test) the 99%ile response time is below 100ms (often reaching 30ms, a 100xish improvement). Yet if we go back to browsing through different pages again, the large queries each take a few seconds to complete.

**To Reproduce**
I currently haven't got an easily sharable test case as it's in a DB with customer data, but I can share some of the GQL queries as examples of what we're running. (These queries are dynamically generated and can definitely be optimised better, but in the load tests below they've proven that they're still able to give speedy results)

<details>
  <summary>Query 1</summary>

  ```GQL
query loadSingleWareneingangsbuchData($id: BigInt) {
  rows: custWareneingangsbuchCollection(filter: {id: {eq: $id}}) {
    edges {
      node {
        id
        zeugnis_vorhanden: zeugnisVorhanden
        waz_bemerkung: wazBemerkung
        waz_art_id: wazArt {
          id
          label
          __typename
        }
        waz_nr: wazNr
        schmelznr
        chemistry_validator: custWareneingangsbuchWerkstoffzustaendeCollection {
          edges {
            node {
              id
              zustand {
                id
                werkstoff_spezifikation: werkstoffSpezifikation {
                  id
                  werkstoff_spezifikation_element: custWerkstoffSpezifikationElementCollection {
                    edges {
                      node {
                        id
                        min
                        max
                        elemente {
                          id
                          name
                          symbol
                          __typename
                        }
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        chemistry_validator_values: custWareneingangsbuchElementeCollection {
          edges {
            node {
              value
              elementeId
              __typename
            }
            __typename
          }
          __typename
        }
        werkstoffzustaende_pivot: custWareneingangsbuchWerkstoffzustaendeCollection(
          orderBy: [{zustandId: AscNullsFirst}]
        ) {
          edges {
            node {
              id
              pivotLabel: zustand {
                id
                werkstoffSpezifikation {
                  werkstoff {
                    label
                    __typename
                  }
                  spezifikation {
                    label
                    __typename
                  }
                  __typename
                }
                zustaende {
                  label
                  __typename
                }
                werkstoff {
                  synonyms
                  __typename
                }
                status {
                  name
                  colour
                  icon
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator: custWareneingangsbuchWerkstoffzustaendeCollection {
          edges {
            node {
              id
              zustand {
                id
                zustand_me: custZustandMeCollection {
                  edges {
                    node {
                      id
                      min
                      max
                      mechanische_eigenschaften: mechanischeEigenschaften {
                        id
                        symbol
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator_values: custWareneingangsbuchMeCollection {
          edges {
            node {
              value
              mechanischeEigenschaftenId
              __typename
            }
            __typename
          }
          __typename
        }
        gepr_gg_spec_id: geprGgSpec {
          id
          label
          __typename
        }
        werkstoffprufunge_pivot: custWareneingangsbuchWerkstoffprufungeCollection(
          orderBy: [{werkstoffprufungeId: AscNullsFirst}]
        ) {
          edges {
            node {
              id
              pivotLabel: werkstoffprufunge {
                id
                label
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        freigegaben
        __typename
      }
      __typename
    }
    __typename
  }
}
```
</details>

<details>
  <summary>Query 2</summary>

  ```GQL
query loadSingleWareneingangsbuchData($id: BigInt) {
  rows: custWareneingangsbuchCollection(filter: {id: {eq: $id}}) {
    edges {
      node {
        id
        referenztest
        wareneingangspruefung
        wareneingangsauftrag
        chemistry_validator_prufung: custWareneingangsbuchWerkstoffzustaendeCollection {
          edges {
            node {
              id
              zustand {
                id
                werkstoff_spezifikation: werkstoffSpezifikation {
                  id
                  werkstoff_spezifikation_element: custWerkstoffSpezifikationElementCollection {
                    edges {
                      node {
                        id
                        min
                        max
                        elemente {
                          id
                          name
                          symbol
                          __typename
                        }
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        chemistry_validator_prufung_values: custWareneingangsbuchElementePrufungCollection {
          edges {
            node {
              value
              elementeId
              __typename
            }
            __typename
          }
          __typename
        }
        werkstoffzustaende_pivot: custWareneingangsbuchWerkstoffzustaendeCollection(
          orderBy: [{zustandId: AscNullsFirst}]
        ) {
          edges {
            node {
              id
              pivotLabel: zustand {
                id
                werkstoffSpezifikation {
                  werkstoff {
                    label
                    __typename
                  }
                  spezifikation {
                    label
                    __typename
                  }
                  __typename
                }
                zustaende {
                  label
                  __typename
                }
                werkstoff {
                  synonyms
                  __typename
                }
                status {
                  name
                  colour
                  icon
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator_prufung: custWareneingangsbuchWerkstoffzustaendeCollection {
          edges {
            node {
              id
              zustand {
                id
                zustand_me: custZustandMeCollection {
                  edges {
                    node {
                      id
                      min
                      max
                      mechanische_eigenschaften: mechanischeEigenschaften {
                        id
                        symbol
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator_prufung_values: custWareneingangsbuchMePrufungCollection {
          edges {
            node {
              value
              mechanischeEigenschaftenId
              __typename
            }
            __typename
          }
          __typename
        }
        werkstoffzustaende_prufung_pivot: custWareneingangsbuchWerkstoffzustaendePrufungCollection(
          orderBy: [{zustandId: AscNullsFirst}]
        ) {
          edges {
            node {
              id
              pivotLabel: zustand {
                id
                werkstoffSpezifikation {
                  werkstoff {
                    label
                    __typename
                  }
                  spezifikation {
                    label
                    __typename
                  }
                  __typename
                }
                zustaende {
                  label
                  __typename
                }
                werkstoff {
                  synonyms
                  __typename
                }
                status {
                  name
                  colour
                  icon
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        gefugeuntersuchung_id: gefugeuntersuchung {
          id
          label
          __typename
        }
        me_validator_referenztest: custWareneingangsbuchWerkstoffzustaendePrufungCollection {
          edges {
            node {
              id
              zustand {
                id
                zustand_me: custZustandMeCollection {
                  edges {
                    node {
                      id
                      min
                      max
                      mechanische_eigenschaften: mechanischeEigenschaften {
                        id
                        symbol
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator_referenztest_values: custWareneingangsbuchMeReferenztestCollection {
          edges {
            node {
              value
              mechanischeEigenschaftenId
              __typename
            }
            __typename
          }
          __typename
        }
        freigegaben
        __typename
      }
      __typename
    }
    __typename
  }
}
```
</details>

<details>
  <summary>Query 3</summary>

  ```GQL
query loadWareneingangsbuchRowData($id: String) {
  rows: custWareneingangsbuchCollection(filter: {id: {eq: $id}}) {
    edges {
      node {
        id
        chargen_nummer: chargenNummer
        werkstoff {
          label
          __typename
        }
        freigegaben
        referenztest
        status_id: status {
          id
          name
          colour
          icon
          __typename
        }
        zeugnis_vorhanden: zeugnisVorhanden
        werkstoff_id: werkstoff {
          id
          label
          status {
            name
            colour
            icon
            __typename
          }
          __typename
        }
        werkstoffzustaende_pivot: custWareneingangsbuchWerkstoffzustaendeCollection(
          orderBy: [{zustandId: AscNullsFirst}]
        ) {
          edges {
            node {
              id
              pivotLabel: zustand {
                id
                werkstoffSpezifikation {
                  werkstoff {
                    label
                    __typename
                  }
                  spezifikation {
                    label
                    __typename
                  }
                  __typename
                }
                zustaende {
                  label
                  __typename
                }
                werkstoff {
                  synonyms
                  __typename
                }
                status {
                  name
                  colour
                  icon
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        abmessung
        abmessungDimension {
          id
          name
          dimension {
            unit
            __typename
          }
          __typename
        }
        waz_bemerkung: wazBemerkung
        bestelldatum_nummer: bestelldatumNummer
        lieferant_id: lieferantId
        liefermenge
        liefermengeDimension {
          id
          name
          dimension {
            unit
            __typename
          }
          __typename
        }
        eingangsdatum
        waz_art_id: wazArt {
          id
          label
          __typename
        }
        waz_nr: wazNr
        schmelznr
        chemistry_validator: custWareneingangsbuchWerkstoffzustaendeCollection {
          edges {
            node {
              id
              zustand {
                id
                werkstoff_spezifikation: werkstoffSpezifikation {
                  id
                  werkstoff_spezifikation_element: custWerkstoffSpezifikationElementCollection {
                    edges {
                      node {
                        id
                        min
                        max
                        elemente {
                          id
                          name
                          symbol
                          __typename
                        }
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        chemistry_validator_values: custWareneingangsbuchElementeCollection {
          edges {
            node {
              value
              elementeId
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator: custWareneingangsbuchWerkstoffzustaendeCollection {
          edges {
            node {
              id
              zustand {
                id
                zustand_me: custZustandMeCollection {
                  edges {
                    node {
                      id
                      min
                      max
                      mechanische_eigenschaften: mechanischeEigenschaften {
                        id
                        symbol
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator_values: custWareneingangsbuchMeCollection {
          edges {
            node {
              value
              mechanischeEigenschaftenId
              __typename
            }
            __typename
          }
          __typename
        }
        gepr_gg_spec_id: geprGgSpec {
          id
          label
          __typename
        }
        prufung
        wareneingangspruefung
        wareneingangsauftrag
        werkstoffprufunge_pivot: custWareneingangsbuchWerkstoffprufungeCollection(
          orderBy: [{werkstoffprufungeId: AscNullsFirst}]
        ) {
          edges {
            node {
              id
              pivotLabel: werkstoffprufunge {
                id
                label
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        ultraschall_gepruft: ultraschallGepruft
        chemistry_validator_prufung: custWareneingangsbuchWerkstoffzustaendeCollection {
          edges {
            node {
              id
              zustand {
                id
                werkstoff_spezifikation: werkstoffSpezifikation {
                  id
                  werkstoff_spezifikation_element: custWerkstoffSpezifikationElementCollection {
                    edges {
                      node {
                        id
                        min
                        max
                        elemente {
                          id
                          name
                          symbol
                          __typename
                        }
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        chemistry_validator_prufung_values: custWareneingangsbuchElementePrufungCollection {
          edges {
            node {
              value
              elementeId
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator_prufung: custWareneingangsbuchWerkstoffzustaendeCollection {
          edges {
            node {
              id
              zustand {
                id
                zustand_me: custZustandMeCollection {
                  edges {
                    node {
                      id
                      min
                      max
                      mechanische_eigenschaften: mechanischeEigenschaften {
                        id
                        symbol
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator_prufung_values: custWareneingangsbuchMePrufungCollection {
          edges {
            node {
              value
              mechanischeEigenschaftenId
              __typename
            }
            __typename
          }
          __typename
        }
        werkstoffzustaende_prufung_pivot: custWareneingangsbuchWerkstoffzustaendePrufungCollection(
          orderBy: [{zustandId: AscNullsFirst}]
        ) {
          edges {
            node {
              id
              pivotLabel: zustand {
                id
                werkstoffSpezifikation {
                  werkstoff {
                    label
                    __typename
                  }
                  spezifikation {
                    label
                    __typename
                  }
                  __typename
                }
                zustaende {
                  label
                  __typename
                }
                werkstoff {
                  synonyms
                  __typename
                }
                status {
                  name
                  colour
                  icon
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        gefugeuntersuchung_id: gefugeuntersuchung {
          id
          label
          __typename
        }
        freigabe_datum: freigabeDatum
        bemerkungen
        createdAt
        updatedAt
        me_validator_referenztest: custWareneingangsbuchWerkstoffzustaendePrufungCollection {
          edges {
            node {
              id
              zustand {
                id
                zustand_me: custZustandMeCollection {
                  edges {
                    node {
                      id
                      min
                      max
                      mechanische_eigenschaften: mechanischeEigenschaften {
                        id
                        symbol
                        __typename
                      }
                      __typename
                    }
                    __typename
                  }
                  __typename
                }
                __typename
              }
              __typename
            }
            __typename
          }
          __typename
        }
        me_validator_referenztest_values: custWareneingangsbuchMeReferenztestCollection {
          edges {
            node {
              value
              mechanischeEigenschaftenId
              __typename
            }
            __typename
          }
          __typename
        }
        lieferung
        materialbeistellung
        freigegaben_von: freigegabenVon
        __typename
      }
      __typename
    }
    pageInfo {
      startCursor
      endCursor
      hasNextPage
      hasPreviousPage
      __typename
    }
    totalCount
    __typename
  }
}
```
</details>

**Expected behavior**
Queries do not slow down exponentially between runs where a user executes other queries.

**Screenshots**
One load test example (hitting the /graphql/v1 endpoint locally with the queries above + a few others):

* On startup the request times are very slow (the max time shown), but after about 10s they normalise to the point where the 99%ile is under 100ms
* The 99%ile continues to drop slowly towards being under 50ms which I feel implies most/all requests continue to be as fast as we hope
* If I start using the application as this is going on the pages that use these queries resolve super fast, but if I start going to pages which send different GQL queries the load test will start seeing slow requests again

image

**Versions:**
- PostgreSQL: 14.1 (Supabase SaaS), 14.2 (cli)
- pg_graphql commit ref: (unsure, I've been testing this today on the Supabase SaaS and Supabase CLI v0.25.0)

**Additional context**
I've tried getting the SQL generated by pg_graphql so I can see if it's an issue with the query itself/something in Postgres, but I could not find a simple way to do this. Is there a way to pull out the generated SQL query from pg_graphql?

From my experimenting, it feels like running more than a few different GQL queries clears old queries out of the cache. Then when the old queries are executed again, the SQL is getting regenereted as it's not longer cached. If so, is there a way to increase the cache size? Our application has a lot of unique queries and the number of these will only grow over time. I may also have misunderstood what is being cached.

I hope that this is enough information, if not I will do what I can to provide more context. Thank you in advance for the help and for making this extension!

### LINKED ISSUE #202: Is there a { returning: "minimal" } option for mutations on tables with RLS with only insert allowed
# Summary
I have a basic table with an email column where I enabled RLS to only allow `INSERT` for `anon`. This is for a use case like a waitlist where the user can add their email to a waitlist but shouldn't be able to query or `SELECT` any of the waitlist.

I noticed [here ](https://github.com/supabase/supabase/issues/1806#issuecomment-854174817)that there's a `{ returning: "minimal" }` option when doing an insert using supabase-js but I didn't see anywhere where that option could be set for a graphql mutation. As a result, I get the error: `new row violates row-level security policy for table`

Is there a way to do this for graphql mutations or is there a recommended alternative method of doing something like this?

### LINKED ISSUE #203: Postgres Enum Types display as "String" in Introspection
Test case:
https://github.com/supabase/pg_graphql/blob/master/test/sql/resolve_graphiql_schema.sql

Output for the enum column
```json
                         {                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  +
                             "args": [                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      +
                             ],                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             +
                             "name": "status",                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              +
                             "type": {                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      +
                                 "kind": "NON_NULL",                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        +
                                 "name": null,                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              +
                                 "ofType": {                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                +
                                     "kind": "SCALAR",                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      +
                                     "name": "String",                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      +
                                     "ofType": null                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         +
                                 }                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          +
                             },                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             +
                             "description": null,                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           +
                             "isDeprecated": false,                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         +
                             "deprecationReason": null                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      +
                         },  
```

### LINKED ISSUE #204: Reflect Filter types for user defined enums

### LINKED ISSUE #207: pg_graphql  does not generate schema for array typed properties on mutation.
**Describe the bug**

Whenever I use the codegen to create the types from the queries it does not generate the properties with [Int] for the update type:

For example, the type "Profile" is correctly generated as:
```
type profiles {
    birthday: Date
    can_be_roommate: Boolean
    created_at: Datetime
    description: String
    display_name: String!
    employment: Int
    first_name: String!
    id: BigInt!
    last_name: String!
    search_as: Int
    tenant_attributes: [Int]
    user_id: UUID!
}
```

But its mutation type is missing the tenant_attributes

```
input profilesUpdateInput {
    birthday: Date
    can_be_roommate: Boolean
    created_at: Datetime
    description: String
    display_name: String
    employment: Int
    first_name: String
    last_name: String
    search_as: Int
    user_id: UUID
}
```

Error:
    × GraphQL Document Validation failed with 1 errors;
      Error 0: GraphQLDocumentError: Field "tenant_attributes" is not defined by type "profilesUpdateInput".
Image

**To Reproduce**
1. Create a table with an array property. Eg. Int[]
2. Add graphql plugin to table/db 
3. Get the schema via. the graphql endpoint
4. Notice the update type

**Expected behavior**
It should also include the array properties in the mutation types

**Versions:**
 - PostgreSQL: Supabase
 - pg_graphql: Whatever is on supabase

### PR DESCRIPTION
**Pull request by olirice**
## What kind of change does this PR introduce?
Ports pg_graphql from SQL to rust

Tasks:
- [x] performance comparison
- [x] tests running in CI
- [x] update installation docs

### Performance
4-5x increase in throughput for on-index simple queries. This is the worse case improvement. The overhead reduction is more significant on large queries that use fragments.

Query
```graphql
{ accountCollection(first: 1) { edges { node { id } } } }
```

#### v0.4.0
```
Concurrency Level:      20
Time taken for tests:   2.827 seconds
Complete requests:      5000
Failed requests:        0
Non-2xx responses:      5000
Total transferred:      2170000 bytes
Total body sent:        1160000
HTML transferred:       1445000 bytes
Requests per second:    1768.94 [#/sec] (mean)
Time per request:       11.306 [ms] (mean)
Time per request:       0.565 [ms] (mean, across all concurrent requests)
Transfer rate:          749.73 [Kbytes/sec] received
                        400.77 kb/s sent
                        1150.50 kb/s total

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    0   0.2      0       1
Processing:     3   11   6.0      9      53
Waiting:        2   11   5.9      9      53
Total:          3   11   6.0      9      53

Percentage of the requests served within a certain time (ms)
  50%      9
  66%     12
  75%     13
  80%     14
  90%     19
  95%     25
  98%     30
  99%     32
 100%     53 (longest request)
```

#### v0.5.0 (this PR)
```
Concurrency Level:      20
Time taken for tests:   0.932 seconds
Complete requests:      5000
Failed requests:        0
Total transferred:      1080000 bytes
Total body sent:        1160000
HTML transferred:       325000 bytes
Requests per second:    5366.30 [#/sec] (mean)
Time per request:       3.727 [ms] (mean)
Time per request:       0.186 [ms] (mean, across all concurrent requests)
Transfer rate:          1131.95 [Kbytes/sec] received
                        1215.80 kb/s sent
                        2347.76 kb/s total

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    0   0.1      0       1
Processing:     1    4   3.9      3      69
Waiting:        1    4   3.9      3      69
Total:          1    4   3.9      3      69

Percentage of the requests served within a certain time (ms)
  50%      3
  66%      4
  75%      4
  80%      4
  90%      5
  95%      7
  98%     16
  99%     25
 100%     69 (longest request)
```

----
resolves #133 
resolves #207
resolves #204
resolves #203 
resolves #202 -- returning clause restricts to selectable columns rather than `*`
resolves #183 -- query compilation incurs no lookup penalty
resolves #170 -- added [test](https://github.com/supabase/pg_graphql/pull/221#discussion_r997321869)

### CODE REVIEW

File: test/expected/resolve_connection_pagination_args.out [hunk 1]
```diff
@@ -370,6 +370,44 @@ begin;
...
+                     "node": {  +
+                         "id": 2+
+                     }          +
+                 }              +
+             ]                  +
+         }                      +
+     }                          +
+ }
+(1 row)
+
```
Comment (olirice) [inline test/expected/resolve_connection_pagination_args.out:410]: test for issue #161

File: test/expected/resolve___schema.out [hunk 2]
```diff
@@ -237,10 +245,6 @@ begin;
                      "kind": "SCALAR",               +
                      "name": "JSON"                  +
                  },                                  +
-                 {                                   +
-                     "kind": "INPUT_OBJECT",         +
-                     "name": "JSONFilter"            +
-                 },                                  +
```
Comment (olirice) [inline test/expected/resolve___schema.out:243]: Removing this type is a user facing breaking change. However, using `JSONFilter` resulted in an error 100% of the time, so no messaging is required around its removal.

**Comment by Flambe** [comment]
This looks really cool!

You said that you're expecting an improvement in the performance of simple queries which is really cool, would this port also improve the schema generation performance? (somewhat related to #220)

**Comment by Flambe** [comment]
> @Flambe the schema will continue to rebuild at the same frequency but the rebuild speed is orders of magnitude faster. You can estimate rebuild time by running an explain plan on the [schema loading query](https://github.com/supabase/pg_graphql/blob/052f93c74a16e177bdb99c30d53f40b97b76f355/sql/load_sql_context.sql) replacing `graphql.get_schema_version` with `1`

Ah sorry for the confusion I meant speed wise.

My local usually takes ~5-10s on the current rebuild schema command. I tried that estimate and it's taking <300ms which is an insane improvement!

Really looking forward to seeing this released :)

**Review by kiwicopple — APPROVED** [review]

**Pull request merged**